static/js/components/PersonalTab.js

```diff
@@ -25,86 +24,80 @@ import type {
     currentProgramEnrollment: AvailableProgram,
+    errors:                   ValidationErrors,
+    nextStep:                 () => void,
+    prevStep:                 () => void,
+    profile:                  Profile,
+    programs:                 AvailablePrograms,
+    saveProfile:              SaveProfileFunc,
+    setProgram:               Function,
+    ui:                       UIState,
+    updateProfile:            UpdateProfileFunc,
   };
 
-  programListing = (programs: AvailablePrograms) => {
-    const makeMenuItems = R.map(program => (
-      <MenuItem value={program.id} key={program.id} primaryText={program.title} />
-    ));
-    const sortPrograms = R.sortBy(R.compose(R.toLower, R.prop('title')));
-    const menuItems = R.compose(makeMenuItems, sortPrograms);
-    return menuItems(programs);
-  };
+  sortPrograms = R.sortBy(R.compose(R.toLower, R.prop('title')));
+
+  programOptions = R.compose(
+    R.map(program => ({value: program.id, label: program.title})), this.sortPrograms
+  );
 
-  setProgramHelper = (event: Event, key: string, value: string) => {
+  onChange = (selection: Option): void => {
     const {
       programs,
       setProgram,
     } = this.props;
-    let selected = programs.find(program => program.id === parseInt(value));
-    setProgram(selected);
+    if ( selection && selection.value ) {
+      let selected = programs.find(program => program.id === parseInt(selection.value));
+      setProgram(selected);
+    } else {
+      setProgram("");
+    }
   };
 
-  getSelectedProgramId = () : number|null => {
+  getSelectedProgramId = (): number|null => {
     const {
       ui: { selectedProgram },
       currentProgramEnrollment
     } = this.props;
-
     let programId = null;
     if (selectedProgram) {
       programId = selectedProgram.id;
     } else if(currentProgramEnrollment) {
       programId = currentProgramEnrollment.id;
     }
-
     return programId;
-  }
-
-  selectProgram = () => {
-    const {
-      programs,
-      errors
-    } = this.props;
+  };
 
+  selectProgram = (): React$Element<*> => {
+    const { programs, errors } = this.props;
     return (
-      <SelectField
+      <Select
         value={this.getSelectedProgramId()}
-        style={{width: "65%"}}
-        hintText="Select Program"
-        onChange={this.setProgramHelper}
+        clearable={false}
+        onChange={this.onChange}
         className={`program-selectfield ${validationErrorSelector(errors, ['program'])}`}
         errorText={_.get(errors, "program")}
-      >
-        { this.programListing(programs) }
-      </SelectField>
+        options={this.programOptions(programs)}
+      />
     );
-  }
+  };
 
-  componentWillMount() {
+  componentDidUpdate() {
     const {
       programs,
       currentProgramEnrollment,
+      ui: { selectedProgram },
       setProgram,
     } = this.props;
 
-    if (currentProgramEnrollment) {
+    if (programs && currentProgramEnrollment && !selectedProgram) {
```

Comment:
I think `programs` always be some kind of array, and arrays are always truthy so we can remove that part of the expression